### PR TITLE
FIT-32 Create DELETE /product endpoint

### PR DESCRIPTION
**Pull request by PiotrMichalak1**

### CODE REVIEW

File: src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java [hunk 1]
```diff
@@ -99,27 +101,45 @@ void givenValidGetRequestWithNonexistentId_shouldReturnError() throws Exception
             assertThat(response).isEqualTo(expectedResponse);
         }
 
+        @Test
+        void givenGetRequestWithExistingIdAndActiveFalse_shouldReturnError() throws Exception {
+            deactivateExisting();
```
[the remark this answers is not in the document]
Comment (PiotrMichalak1) [reply src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java:106]: As in previous one, I'm concerned about coupling.

File: src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java [hunk 2]
```diff
@@ -48,9 +51,12 @@ protected String getEndpoint() {
         return ENDPOINT;
     }
 
+    private UUID id;
+
     @BeforeEach
     void beforeEach() {
         productRepository.deleteAll();
+        id = productRepository.save(product()).getId();
```
[the remark this answers is not in the document]
Comment (pd-dudkiewicz) [reply src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java]: Seems reasonable, let's leave it as it was then

File: src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pd-dudkiewicz) [reply src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java]: We test DELETE separately so we can assume it's working as expected and it's fine to use it in other tests

File: src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java [hunk 3]
```diff
@@ -63,34 +69,30 @@ void givenValidPostRequest_shouldSaveAndReturnProduct() throws Exception {
...
         @Test
-        void givenValidGetRequestWithExistingId_shouldReturnProduct() throws Exception {
-            productRepository.save(product());
-
-            UUID productUuid = productRepository.findAll().iterator().next().getId();
-            var expectedResponse = productResponse();
-            var response = makeRequest(ENDPOINT + "/" + productUuid, GET, OK, ProductResponse.class);
+        void givenGetRequestWithExistingIdAndActiveTrue_shouldReturnActiveProduct() throws Exception {
+            deactivateExisting();
+            addNewVersion();
```
[the remark this answers is not in the document]
Comment (PiotrMichalak1) [reply src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java]: To ensure that only active version of a product is returned.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 target/
 .idea
 .DS_Store
+drafts/
```
[the remark this answers is not in the document]
Comment (PiotrMichalak1) [reply .gitignore:4]: Folder that I have created to hold some temporary files, to not use a notepad.

**Review by pd-dudkiewicz — COMMENTED** [review]
You've got `Assertions.assertThat` in a few places instead of static imports. Other than that it looks good, you can squash commits once you've fixed the imports and we can merge.

**Review by pd-dudkiewicz — APPROVED** [review]

**Pull request merged**